themes/V3/5ePHB/snippets/tableOfContents.gen.js

```diff
@@ -2,77 +2,66 @@ const _ = require('lodash');
 const dedent = require('dedent-tabs').default;
 
 const getTOC = (pages)=>{
-	const add1 = (title, page)=>{
-		res.push({
-			title    : title,
-			page     : page + 1,
-			children : []
-		});
-	};
-	const add2 = (title, page)=>{
-		if(!_.last(res)) add1(null, page);
-		_.last(res).children.push({
-			title    : title,
-			page     : page + 1,
-			children : []
-		});
-	};
-	const add3 = (title, page)=>{
-		if(!_.last(res)) add1(null, page);
-		if(!_.last(_.last(res).children)) add2(null, page);
-		_.last(_.last(res).children).children.push({
-			title    : title,
-			page     : page + 1,
-			children : []
-		});
+
+	const recursiveAdd = (title, page, targetDepth, child, curDepth=0)=>{
+		if(curDepth > 5) return; // Something went wrong.
+		if(curDepth == targetDepth) {
+			child.push({
+				title    : title,
+				page     : page + 1,
+				children : []
+			});
+		} else {
+			child.push({
+				title    : null,
+				page     : page + 1,
+				children : []
+			});
+			recursiveAdd(title, page, targetDepth, _.last(child).children, curDepth+1,);
+		}
 	};
 
 	const res = [];
-	_.each(pages, (page, pageNum)=>{
-		if(!page.includes('{{frontCover}}') && !page.includes('{{insideCover}}') && !page.includes('{{partCover}}') && !page.includes('{{backCover}}')) {
-			const lines = page.split('\n');
-			_.each(lines, (line)=>{
-				if(_.startsWith(line, '# ')){
-					const title = line.replace('# ', '');
-					add1(title, pageNum);
-				}
-				if(_.startsWith(line, '## ')){
-					const title = line.replace('## ', '');
-					add2(title, pageNum);
-				}
-				if(_.startsWith(line, '### ')){
-					const title = line.replace('### ', '');
-					add3(title, pageNum);
-				}
-			});
+
+	const iframe = document.getElementById('BrewRenderer');
+	const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
+	const headings = iframeDocument.querySelectorAll('h1, h2, h3, h4, h5, h6');
+	const headerDepth = ['H1', 'H2', 'H3', 'H4', 'H5', 'H6'];
+
+	_.each(headings, (heading)=>{
+		const onPage = parseInt(heading.closest('.page').id?.replace(/^p/, ''));
+		const ToCExclude = getComputedStyle(heading).getPropertyValue('--TOC');
+
+		if(ToCExclude != 'exclude') {
+			recursiveAdd(heading.innerText.trim(), onPage, headerDepth.indexOf(heading.tagName), res);
 		}
 	});
 	return res;
 };
 
+
+const ToCIterate = (entries, curDepth=0)=>{
+	const levelPad = ['- ###', '  - ####', '    - ####', '      - ####', '        - ####', '          - ####'];
```

Comment:
After The first two levels, any other nested ToC levels should have no `#` attached (displayed as plain text)

```suggestion
	const levelPad = ['- ###', '  - ####', '    - ', '      - ', '        - ', '          - '];
```

This change would get us mostly there, but there is one more catch. Indentation level also depends on the previous indentation level. So an H3 following an H1 should only be indented once.

See this example for what we are looking for (as implemented in the current live site). The H3 entries "Homebrew D&D made easy" and "Editing and Sharing" are only indented once since there is no intermediate H2. However the later H3 entries "Helping out", "Bugs, Issues, Suggestions?", etc. do have an intermediary H2, so they should be indented twice.

![image](https://github.com/naturalcrit/homebrewery/assets/5878534/699e5f47-b0e3-470a-ae73-7ae2cfa2108d)